torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -497,6 +497,109 @@ TensorView* TensorView::rFactor(const std::vector<int>& axes) {
   return producer;
 }
 
+std::vector<TensorView*> TensorView::rFactorWelford(
+    const std::vector<int>& axes) {
+  TORCH_INTERNAL_ASSERT(nDims() > 0, "Tried to rFactor a 0-dim TensorView");
+  FusionGuard fg(fusion());
+  TORCH_CHECK(
+      definition() != nullptr &&
+          definition()->getExprType() == ExprType::WelfordOp,
+      "Error rfactoring welford ",
+      this,
+      " its definition is either a nullptr or not a welford.");
+  TORCH_CHECK(
+      !domain()->hasRFactor(), "Cannot call rfactor on the same view twice.");
+
+  WelfordOp* wop = definition()->as<WelfordOp>();
+
+  TensorView* this_var = wop->outVar()->as<TensorView>();
+  TensorView* this_avg = wop->outAvg()->as<TensorView>();
+  TensorView* this_n = wop->outN()->as<TensorView>();
+
+  auto rfactorTVHelper = [&](TensorView* tv) -> TensorView* {
```

Comment:
This lambda is longer than the rest of the function body. Wouldn't it make more sense to make it a separate function?

Reply:
Moved these logic to `TensorView::WelfordRfactorHelper`. It was a lambda function within `TensorView` because the logic needed to access private methods in `TensorView`. I guess making it another private method of `TensorView` is cleaner. Thanks for the suggestion.